tweakwcs/imalign.py

```diff
@@ -282,7 +282,7 @@ def tweak_image_wcs(images, refcat=None, enforce_user_order=True,
 
             rcat = refcat.meta['catalog'].copy()
 
-            if 'RA' not in catalog.colnames or 'DEC' not in catalog.colnames:
+            if 'RA' not in refcat.colnames or 'DEC' not in refcat.colnames:
```

Comment:
Where rcat is just a copy, this validation step works equally for both objects.  However, if you feel it needs to be modified, I can do that....

Reply:
In this line: https://github.com/spacetelescope/tweakwcs/blob/0aeaf8e5c50b08542f889aa42dffa5fa0052916f/tweakwcs/imalign.py#L278 it seems that it is assured that `refcat` is an `NDData` object. It usually does not have `colnames` attribute. That is why I do not think this can work.

Reply:
Actually, I have been running this code perfectly successfully this afternoon when I provide a QTable/Table object as input.

Reply:
When `refcat` is an `astropy.table.Table`, code goes through this branch: https://github.com/spacetelescope/tweakwcs/blob/0aeaf8e5c50b08542f889aa42dffa5fa0052916f/tweakwcs/imalign.py#L306

which is seemingly (thanks to your PR) buggy (`catalog`->`refcat`) so I can't understand why did it work at all... I wonder if your reference catalog was of supported type. It looks like the `if` statement dooes not have an `else` branch to raise an exception.

Reply:
My test confirms my suspicion that your fix does not fork for `NDData` objects. For example, for JWST data models, I get:
```
AttributeError: No attribute 'colnames'
```